internal/sys/fs.go

```diff
@@ -420,11 +392,13 @@ func (c *FSContext) CloseFile(fd uint32) bool {
 // Close implements api.Closer
 func (c *FSContext) Close(context.Context) (err error) {
 	// Close any files opened in this context
-	for fd, entry := range c.openedFiles {
-		delete(c.openedFiles, fd)
+	c.openedFiles.Range(func(fd uint32, entry *FileEntry) bool {
 		if e := entry.File.Close(); e != nil {
 			err = e // This means err returned == the last non-nil error.
 		}
-	}
+		return true
+	})
+	// TODO: Required to pass TestEmptyFSContext but Reset would be better.
```

Comment:
a closed context can't be re-opened, why would we have a TODO to use reset?

Reply:
I assumed it could, but if it cannot there's no need for this comment indeed 👍